src/gc/env/gcenv.base.h

```diff
@@ -281,6 +283,62 @@ inline uint8_t BitScanForward64(uint32_t *bitIndex, uint64_t mask)
 #endif // _MSC_VER
 }
 
+// Cross-platform wrapper for the _BitScanReverse compiler intrinsic.
+inline uint8_t BitScanReverse(uint32_t *bitIndex, uint32_t mask)
+{
+#ifdef _MSC_VER
+    return _BitScanReverse((unsigned long*)bitIndex, mask);
+#else // _MSC_VER
+    unsigned char ret = FALSE;
+    if (mask != 0)
+    {
+        int lzcount = __builtin_clzl(mask);
+        *bitIndex = (uint32_t)(31 - lzcount);
+        ret = TRUE;
+    }
+
+    return ret;
+#endif // _MSC_VER
+}
+
+// Cross-platform wrapper for the _BitScanReverse64 compiler intrinsic.
+inline uint8_t BitScanReverse64(uint32_t *bitIndex, uint64_t mask)
+{
+#ifdef _MSC_VER
+ #if _WIN32
+    // MSVC targeting a 32-bit target does not support this intrinsic.
+    // We can fake it using two successive invocations of _BitScanReverse.
+    uint32_t hi = (mask >> 32) & 0xFFFFFFFF;
```

Comment:
Why two invocations rather than just one (by checking if upper is zero or not):
```cpp
uint32_t upper = (uint32_t)(mask >> 32);

if (upper != 0)
{
    uint8_t result = BitScanReverse(bitIndex, upper);
    *bitIndex += 32;
    return result;
}

return BitScanReverse(bitIndex, (uint32_t)(mask));

```

Reply:
Good point! I'll make that change and update the PR.